account_spread_cost_revenue/models/account_spread.py

```diff
@@ -0,0 +1,517 @@
+    )
+    invoice_line_id = fields.Many2one(
+        "account.move.line",
+        string="Invoice line",
+        compute="_compute_invoice_line",
+        inverse="_inverse_invoice_line",
+        store=True,
+    )
+    invoice_id = fields.Many2one(
+        related="invoice_line_id.move_id", readonly=True, store=True,
+    )
+    estimated_amount = fields.Float(digits="Account")
+    company_id = fields.Many2one(
+        "res.company", default=lambda self: self.env.company, required=True
+    )
+    currency_id = fields.Many2one(
+        "res.currency",
+        required=True,
+        default=lambda self: self.env.company.currency_id.id,
+    )
+    account_analytic_id = fields.Many2one(
+        "account.analytic.account", string="Analytic Account"
+    )
+    analytic_tag_ids = fields.Many2many("account.analytic.tag", string="Analytic Tags")
+    move_line_auto_post = fields.Boolean("Auto-post lines", default=True)
+    display_create_all_moves = fields.Boolean(
+        compute="_compute_display_create_all_moves",
+    )
+    display_recompute_buttons = fields.Boolean(
+        compute="_compute_display_recompute_buttons",
+    )
+    display_move_line_auto_post = fields.Boolean(
+        compute="_compute_display_move_line_auto_post",
+        string="Display Button Auto-post lines",
+    )
+    active = fields.Boolean(default=True)
+
+    @api.model
+    def default_journal(self, company_id):
+        domain = [("type", "=", "general"), ("company_id", "=", company_id)]
+        return self.env["account.journal"].search(domain, limit=1)
+
+    @api.model
+    def default_get(self, fields):
+        res = super().default_get(fields)
+        if "journal_id" not in res:
+            company_id = res.get("company_id", self.env.company.id)
+            default_journal = self.default_journal(company_id)
+            if default_journal:
+                res["journal_id"] = default_journal.id
+        return res
+
+    @api.depends("invoice_type")
+    def _compute_spread_type(self):
+        for spread in self:
+            if spread.invoice_type in ["out_invoice", "out_refund"]:
+                spread.spread_type = "sale"
+            else:
+                spread.spread_type = "purchase"
+
+    @api.depends("invoice_line_ids", "invoice_line_ids.move_id")
+    def _compute_invoice_line(self):
+        for spread in self:
+            invoice_lines = spread.invoice_line_ids
+            spread.invoice_line_id = invoice_lines and invoice_lines[0] or False
+
+    def _inverse_invoice_line(self):
+        for spread in self:
+            invoice_line = spread.invoice_line_id
+            spread.write({"invoice_line_ids": [(6, 0, [invoice_line.id])]})
+
+    @api.depends(
+        "estimated_amount",
+        "currency_id",
+        "company_id",
+        "invoice_line_id.price_subtotal",
+        "invoice_line_id.currency_id",
+        "line_ids.amount",
+        "line_ids.move_id.state",
+    )
+    def _compute_amounts(self):
```

Comment:
@astirpe This method does not take into account if the spread is for a credit or a normal invoice. Hence the amounts are calculated incorrectly for i.e. credit notes.

Reply:
Thanks @CasVissers-360ERP. Do you have any solution already implemented for this?